cql3/statements/create_keyspace_statement.cc

```diff
@@ -188,13 +189,39 @@ std::optional<sstring> check_restricted_replication_strategy(
     for (auto opt : attrs.get_replication_options()) {
         try {
             auto rf = std::stol(opt.second);
-            if (rf > 0 && rf < qp.proxy().data_dictionary().get_config().minimum_keyspace_rf()) {
-                throw exceptions::configuration_exception(format(
-                    "Replication factor {}={} is forbidden by the current "
-                    "configuration setting of minimum_keyspace_rf={}. Please "
-                    "increase replication factor, or lower minimum_keyspace_rf "
-                    "set in the configuration.", opt.first, opt.second,
-                    qp.proxy().data_dictionary().get_config().minimum_keyspace_rf()));
+            if (rf > 0) {
+                if (auto min_fail = qp.proxy().data_dictionary().get_config().minimum_replication_factor_fail_threshold();
+                    min_fail >= 0 && rf < min_fail) {
```

Comment:
Nitpick: the min_fail>=0 check isn't needed, because you already know rf > 0, so if min_fail wasn't >=0, you couldn't have rf < min_fail.
Since you don't need to check min_fail >=0 you also don't need to add the min_fail variable, which you only use once.

Reply:
I guess with max_fail below the situation is different, though.

Reply:
Yeah, but because guardrails with values lower than 0 are disabled, for consistency reasons I added everywhere a check that they're >= 0.